catalog/resource_sql_table.go

```diff
@@ -294,29 +294,66 @@ func (ti *SqlTableInfo) buildTableCreateStatement() string {
 	return strings.Join(statements, "")
 }
 
-// Wrapping the column name with backtiks to avoid special character messing things up.
+// Wrapping the column name with backticks to avoid special character messing things up.
 func getWrappedColumnName(ci SqlColumnInfo) string {
 	return fmt.Sprintf("`%s`", ci.Name)
 }
 
 func (ti *SqlTableInfo) getStatementsForColumnDiffs(oldti *SqlTableInfo, statements []string, typestring string) []string {
-	// TODO: take out "force_new" in `column` and add case to handle addition and removal of columns.
-	for i, ci := range ti.ColumnInfos {
-		oldCi := oldti.ColumnInfos[i]
-		if ci.Name != oldCi.Name {
-			statements = append(statements, fmt.Sprintf("ALTER %s %s RENAME COLUMN %s to %s", typestring, ti.SQLFullName(), getWrappedColumnName(oldCi), getWrappedColumnName(ci)))
+	if len(ti.ColumnInfos) != len(oldti.ColumnInfos) {
+		nameToOldColumn := make(map[string]SqlColumnInfo)
```

Comment:
This method is too long & too deeply nested (code smell). Let's split this into a couple separate methods for readability. I feel like the core logic is:
```go
if len(ti.ColumnInfos) != len(oldti.ColumnInfos) {
  return addOrRemoveColumnStatements()
}
return alterExistingColumnStatements()
```

Much easier to read and review + decoupling independent logical components from one another.